### PR TITLE
hardening: remove /bin/sh fallback from alert command execution

### LINKED ISSUE #257: hardening: remove /bin/sh fallback from alert command execution path
## Summary
Alert command execution currently falls back to shell invocation (`/bin/sh`) for non-executable first token.

## Evidence
`functions.php`:
- around `1523-1534`
- around `1583-1594`

Behavior:
- if first token is executable: `exec($command, ...)`
- else: `exec('/bin/sh ' . $command, ...)`

## Risk
Expands command-injection/blast-radius profile and makes execution semantics harder to reason about.

## Expected fix
- Remove shell fallback path
- Execute only explicitly allowed executable/script targets
- Prefer fixed argv construction over shell command strings
- Surface clear error when command is invalid

### PR DESCRIPTION
**Pull request by somethingwithproof**
## Summary
- remove /bin/sh fallback execution path for alert commands
- execute commands only when the first token resolves to an executable file
- log explicit errors and return code 126 for non-executable command targets
- add regression test ensuring shell fallback is not reintroduced
- update changelog with issue #257

Fixes #257

## Validation
- php -l functions.php
- php -l tests/regression/issue257_remove_shell_fallback_test.php
- php tests/regression/issue257_remove_shell_fallback_test.php

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -1585,12 +1588,15 @@ function syslog_process_alert($alert, $sql, $params, $count, $hostname = '') {
 
 						$logMessage = "SYSLOG NOTICE: Executing '$command'";
 
-						$cparts = explode(' ', $command);
+						$cparts     = preg_split('/\s+/', trim($command));
+						$output     = array();
+						$returnCode = 0;
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Fixed — log moved inside the is_executable branch.

File: functions.php [hunk 2]
```diff
@@ -1525,12 +1525,15 @@ function syslog_process_alert($alert, $sql, $params, $count, $hostname = '') {
...
 
-						if (is_executable($cparts[0])) {
+						if (cacti_sizeof($cparts) && is_executable($cparts[0])) {
 							exec($command, $output, $returnCode);
 						} else {
-							exec('/bin/sh ' . $command, $output, $returnCode);
+							$returnCode = 126;
+							cacti_log("SYSLOG ERROR: Alert command is not executable: '$command'", false, 'SYSTEM');
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Fixed -- error message now distinguishes PATH-based lookups (unsupported) from missing/non-executable files.

[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Cacti alert commands are stored as plain absolute paths without quotes; the trim() of quote chars handles edge cases but quoted-with-spaces paths are not a supported format in the UI.

[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Fixed -- the else branch already emits 'PATH-based lookups are not supported; use an absolute path' to distinguish this case.

File: functions.php [hunk 3]
```diff
@@ -1585,12 +1588,15 @@ function syslog_process_alert($alert, $sql, $params, $count, $hostname = '') {
 
 						$logMessage = "SYSLOG NOTICE: Executing '$command'";
 
-						$cparts = explode(' ', $command);
+						$cparts     = preg_split('/\s+/', trim($command));
+						$output     = array();
+						$returnCode = 0;
 
-						if (is_executable($cparts[0])) {
+						if (cacti_sizeof($cparts) && is_executable($cparts[0])) {
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Cacti alert commands are stored as plain absolute paths without quotes; the trim() of quote chars handles edge cases but quoted-with-spaces paths are not a supported format in the UI.

File: tests/regression/issue257_remove_shell_fallback_test.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	fwrite(STDERR, "Expected command tokenization update is missing.\n");
+	exit(1);
+}
+
+if (substr_count($functions, '$returnCode = 126;') < 2) {
+	fwrite(STDERR, "Expected non-executable return code handling is missing.\n");
+	exit(1);
+}
+
+if (substr_count($functions, "SYSLOG ERROR: Alert command is not executable: '\$command'") < 2) {
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/regression/issue257_remove_shell_fallback_test.php]: The test verifies the exact code construct that implements the behavior; a content change would indicate a regression.

File: functions.php [hunk 5]
```diff
@@ -1525,12 +1525,15 @@ function syslog_process_alert($alert, $sql, $params, $count, $hostname = '') {
 
 						$logMessage = "SYSLOG NOTICE: Executing '$command'";
 
-						$cparts = explode(' ', $command);
+						$cparts     = preg_split('/\s+/', trim($command));
+						$output     = array();
+						$returnCode = 0;
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Fixed -- log message moved to exec branch only.

[the remark this answers is not in the document]
Comment (somethingwithproof) [reply functions.php]: Fixed — log moved inside the is_executable branch; non-executable path logs SYSLOG ERROR instead.

**Comment by somethingwithproof** [comment]
Closing in favor of #281, which extracts the same exec() hardening into shared helpers (superset of this fix).
